src/auth/user-record.ts

```diff
@@ -173,7 +178,16 @@ export class UserRecord {
       providerData.push(new UserInfo(entry));
     }
     utils.addReadonlyGetter(this, 'providerData', providerData);
-    utils.addReadonlyGetter(this, 'passwordHash', response.passwordHash);
+
+    // If the password hash is redacted (probably due to missing permissions)
+    // then clear it out, similar to how the salt is returned. (Otherwise, it
+    // *looks* like a b64-encoded hash is present, which is confusing.)
+    if (response.passwordHash === B64_REDACTED) {
+      utils.addReadonlyGetter(this, 'passwordHash', undefined);
```

Comment:
Please add a unit test for this. When a user record is initialized with a server response containing a redacted password, confirm it is cleared.

Reply:
Done.

Reply:
Thanks for adding to auth.spec.ts. Can you also add to user-record.spec.ts, the test associated with this class?

Reply:
Done. I've removed the test from auth.spec.ts, as I think with this change, it's become mostly redundant. Though we can add it back in if you'd prefer to keep it.